readline/src/main/java/org/aesh/readline/Buffer.java

```diff
@@ -54,10 +57,12 @@
     private boolean isPromptDisplayed = false;
     private boolean deletingBackward = true;
 
+    private final CursorLocator locator;
```

Comment:
Yes, we could do that.It would add a bit of complexity in order to deal with the null case (jn reset, invalidateCursorLocation and updateMultiLineBuffer).